docs/docsite/rst/lookup_guide.rst

```diff
@@ -0,0 +1,159 @@
+
+So the third thing the lookup does is assume that you're probably trying to read a ``kv`` secret, and to then try to infer if the response is from ``kv2``, because the responses from version 2 include metadata and have the secret value additionally wrappped in another structure. In that way, it seeks to make the response from version 2 look more like the response from version 1.
+
+Since the ``kv`` store has one or more key/value pairs in each secret, the lookup also supports a non-standard suffix in its path that can be used to access a value belonging to one specific key, via the ``:keyname`` syntax. While this is useful to provide a compact way to access a single secret value (admittedly a very common use case), it complicates the implementation and leads to bad habits.
+
+For example, it became common to see people use many lookup invocations with the same path, each with a different ``:keyname``, to access multiple values within a single secret, but this is quite wasteful, as it does a separate login and secret lookup, all to return the same value, and the key dereferencing is done client side. Further, dereferencing can be done directly in Jinja where it's more clear what's going on, using the ``.key`` or ``['key']`` syntax.
+
+One last idiosyncrasy of the plugin is its support for supplying all of its parameters in the term string. This looks compact, but it greatly complicates the processing of plugin options. At the time that this lookup was created, many other lookups allowed options to be supplied in the term string, but it has since been considered an anti-pattern, and has been deprecated/removed from core plugins.
+
+Another downside of this is that it prevents us from effectively re-using the authentication token in cases when multiple term strings are supplied, directly or via ``with_community.hashi_vault.hashi_vault``, and as a result this type of usage results in a new login for each term. In newer lookups, we can take advantage of a single login to perform multiple operations.
+
+All of these considerations make sense in context, but it somewhat muddles the purpose of the lookup:
+
+* If a response from a completely different endpoint ended up looking like a ``kv2`` response, it would return an unexpected result.
+* If you try to give the path of a ``kv2`` secret direcrly, it will not work unless you insert a ``/data/`` component into the path, in order to match the *API path* rather than the path people are usually familiar with.
+* If you want the metadata returned along with a ``kv2`` response, you cannot get it.
+* Other features of ``kv2`` like secret versioning cannot directly be used, unless you modify the URL, which is error prone and unintuitive.
+* Getting access to the token created by the internal login, in order to re-use it, is not possible.
+
+How we are addressing the considerations
```

Comment:
Hmmmm. This is the moment where the doc changes from a description of what the plugin does, to a development roadmap of sorts. If that is the overall intention of the document, then I'd suggest a different main heading at the top and a slightly revised structure.

Reply:
It isn't intended to be a roadmap, only to address the question of this plugin that we will kind of be encouraging folks to consider not using in favor of alternatives.

Reply:
Ah, gotcha.